kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -99,21 +103,33 @@ public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentW
                 CONFLICT_RESOLUTION_SCHEMA,
                 Optional.empty())) {
 
-            actionsIterator.forEachRemaining(actionBatch -> {
+            List<ActionWrapper> actionBatchList = new ArrayList<>();
+            actionsIterator.forEachRemaining(actionBatchList::add);
+            for (int i = 0; i < actionBatchList.size(); i++) {
+                ActionWrapper actionBatch = actionBatchList.get(i);
                 checkArgument(!actionBatch.isFromCheckpoint());  // no checkpoints should be read
                 ColumnarBatch batch = actionBatch.getColumnarBatch();
+                if (i == actionBatchList.size() - 1) {
+                    CommitInfo commitInfo =
+                            getCommitInfo(batch.getColumnVector(COMMITINFO_ORDINAL));
+                    winningCommitInfoOpt = Optional.ofNullable(commitInfo);
+                }
```

Comment:
If the goal is to get the last commit info, why not always look for the commit info, store in a variable, and the last value of the variable is the one we need?

Also, we can't assume the commit info to be present in the last action batch. A commit file could be read could generate multiple batches and the first batch in the list contains the commit info.

Reply:
We couldn't always store commit info in a variable and use the last value of the variable because this [scenario](https://github.com/delta-io/delta/pull/3283#discussion_r1645227258) mentioned by @dhruvarya-db. But you are right, it's not necessarily in the last action batch.

Reply:
Does it make sense to check if a batch is generated from the last commit file by checking its version from ActionWrapper?

Reply:
I also add a [test](https://github.com/delta-io/delta/blob/2748b6357ddb958be49b74ceec933b0b267ac89a/kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala#L490) for this [scenario](https://github.com/delta-io/delta/pull/3283#discussion_r1645227258).

Reply:
Pushed the solution by checking version from ActionWrapper.